### PR TITLE
Feature/cheval command

### PR DESCRIPTION
**Pull request by JohnGilb**

### CODE REVIEW

File: CompetitionTemplate/src/competition/defense_commands/ChevalCommandGroup.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    
+    @Override
+    public void initialize() {
+        backUp.setTargetDistance(chevalBackUpDistance.get());
+        crossDefense.setTargetDistance(chevalCrossDefenseDistance.get());
+        
+        this.addSequential(backUp);
+        this.addSequential(dropArm);
+        this.addSequential(crossDefense);
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/defense_commands/ChevalCommandGroup.java]: I feel like you might want a wait between arm drop and drive forward

File: CompetitionTemplate/src/competition/operator_interface/OperatorCommandMap.java [hunk 2]
```diff
@@ -53,12 +54,15 @@ public void setupDriveCommands(
...
-            ResetRobotPositionCommand resetPosition
+            ResetRobotPositionCommand resetPosition,
+            ChevalCommandGroup cheval
             )
     {
         operatorInterface.leftButtons.getifAvailable(2).whenPressed(calibrateHeading);
         
         resetPosition.includeOnSmartDashboard("Reset Position");
+        
+        operatorInterface.leftButtons.getifAvailable(3).whileHeld(cheval);
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/operator_interface/OperatorCommandMap.java]: While held is a little sketchy here if the command can end (and thus start again). Maybe we should overwrite the isFinished to always say False?

File: CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToDistanceCommand.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import com.google.inject.Inject;
+
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.drive.PoseSubsystem;
+import xbot.common.command.BaseCommand;
+import xbot.common.math.PIDManager;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class DriveToDistanceCommand extends BaseCommand {
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToDistanceCommand.java]: Worth seeing if we can use the one from the low bar feature branch instead (and maybe spruce it up)

Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToDistanceCommand.java]: Ah right, turns out it's in the auto package and is already there

File: CompetitionTemplate/src/competition/defense_commands/ChevalCommandGroup.java (hunk 1 above)
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/defense_commands/ChevalCommandGroup.java]: Good idea. Added a configurable (but default 0.5) second delay between arm drop and drive forward.

File: CompetitionTemplate/src/competition/operator_interface/OperatorCommandMap.java (hunk 2 above)
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/operator_interface/OperatorCommandMap.java]: Nice idea! I created a ChevalCommandThatNeverStops for the OI to use, and the other one can be used during auto.

File: CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToDistanceCommand.java (hunk 3 above)
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/drive/commands/DriveToDistanceCommand.java]: Unified!

**Pull request merged**